### PR TITLE
switch from using NAN to NAPI

### PR DESCRIPTION
**Pull request by ckirmse**
- NAPI gives long-term support
- old code had tight v8 dependencies that do not work with node 12 and later
- also changed to use a newer version of the native code that has better error codes

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [submodule "src/rpi_ws281x"]
     path = src/rpi_ws281x
-    url = https://github.com/beyondscreen/rpi_ws281x.git
+    url = https://github.com/ckirmse/rpi_ws281x.git
```
Comment (usefulthink) [inline .gitmodules:3]: are there any changes to the upstream repo that you need for this? 
Otherwise I'd love to switch back to the upstream version.

**Comment by ckirmse** [comment]
No, the upstream repo is fine!ChrisOn Jul 6, 2020 2:50 AM, Martin Schuhfuss <notifications@github.com> wrote:
@usefulthink approved this pull request.

First of all, sorry for not noticing this earlier and thank you sooooo much for putting the time into this. Was on my list of things to do once I get around to it (I didn't for quite some time as you can tell).
Besides the one question above this all looks perfectly fine.
I can even use your code as a blueprint to also update the newer version of the api.
(once I find the  time, energy and raspberry pis to do that :D)

In .gitmodules:
> @@ -1,3 +1,3 @@
 [submodule "src/rpi_ws281x"]
     path = src/rpi_ws281x
-    url = https://github.com/beyondscreen/rpi_ws281x.git
+    url = https://github.com/ckirmse/rpi_ws281x.git

are there any changes to the upstream repo that you need for this?
Otherwise I'd love to switch back to the upstream version.

File: .gitmodules (hunk 1 above)
Comment (usefulthink) [reply .gitmodules:3]:
```suggestion
    url = https://github.com/jgarff/rpi_ws281x.git
```

**Comment by usefulthink** [comment]
@ckirmse done! I published a new version (0.10.0) to npm with the `@preview`-tag. 

Can you please install that (`npm install rpi-ws281x-native@preview`) and verify that it works?

**Comment by ckirmse** [comment]
OK I did that and it's working great. Nicely done!

**Comment by usefulthink** [comment]
alright! I updated the dist-tags, so you can now install your version as usual with `npm install rpi-ws281x-native`.

Thanks again!

**Pull request merged**